hardware_interface/include/hardware_interface/system_interface.hpp

```diff
@@ -98,9 +100,56 @@ class SystemInterface : public rclcpp_lifecycle::node_interfaces::LifecycleNodeI
   virtual CallbackReturn on_init(const HardwareInfo & hardware_info)
   {
     info_ = hardware_info;
+    add_state_interface_descriptions();
+    add_command_interface_descriptions();
     return CallbackReturn::SUCCESS;
   };
 
+  virtual void add_state_interface_descriptions()
```

Comment:
Method name is not so good. Why don't you use the stateless method, this means doesn't access to the global variables? This parsing will repeat in all components, so maybe adding this into URDF parser or Hardware Info might be enough. Actually, we have this already in the `InterfaceInfo` class. OK, only sorting into different vectors is not there… Perhaps something be added on the parser level, as it is repeatable.

Reply:
Cannot think of a better name. moved it to component parser

Reply:
renamed it to `import _<command/state>_interface_descriptions`